wukong/request.py

```diff
@@ -118,13 +118,9 @@ def request(self, path, params, method, body=None, headers=None, is_retry=False)
 
         response = None
         for host in random.sample(self.master_hosts, len(self.master_hosts)):
-            full_path = urljoin(host, path)
+            full_path = '/'.join(s.strip('/') for s in [host, path])
```

Comment:
Did you switch out `urljoin` because you don't trust it anymore? Personally, I feel like it conveys the intent in a more concise and readable line.

Reply:
It does some hella weird shit too. 
```py
urljoin('http://solr-host:8983/solr', 'collection/select') == 'solr-host:8983/collection/select'  # drops the /solr

urljoin('solr-host:8983', 'collection/select') == 'collection/select'  # drops the entire host
```

I think it's just not the right tool for what we're trying to do. Seems to be made for modifying pieces of the url rather than actually joining together parts of the path like `os.path.join`

Reply:
oh, gross. Makes sense!